src/Momento.Sdk/Exceptions/FailedPreconditionException.cs

```diff
@@ -8,10 +10,8 @@ namespace Momento.Sdk.Exceptions;
 /// </summary>
 public class FailedPreconditionException : SdkException
 {
-    public FailedPreconditionException(string message) : base(MomentoErrorCode.FAILED_PRECONDITION_ERROR, message)
-    {
-    }
-    public FailedPreconditionException(string message, MomentoErrorTransportDetails transportDetails) : base(MomentoErrorCode.FAILED_PRECONDITION_ERROR, message, transportDetails)
+    public FailedPreconditionException(string message, MomentoErrorTransportDetails? transportDetails=null, Exception? e=null) : base(MomentoErrorCode.FAILED_PRECONDITION_ERROR, message, transportDetails, e)
     {
+        this.MessageWrapper = "System is not in a state required for the operation's execution";
```

Comment:
I think we want to pass the server message through on this one too.  AFAIK the only place on the server where we are currently using this is if they try to do an "increment" on a non-string value and the message from the server should hopefully state that pretty clearly.